src/ivc/fold_relaxed_plonk_instance_chip.rs

```diff
@@ -1023,6 +1060,160 @@ mod tests {
         }
     }
 
+    #[test_log::test]
+    fn fold_E_test() {
+        let Fixture {
+            mut td,
+            config,
+            mut rnd,
+            ecc,
+            gate,
+            r,
+        } = Fixture::default();
+
+        let mut folded_E = C1::default();
+
+        let mut layouter = SingleChipLayouter::new(&mut td, vec![]).unwrap();
+
+        let mut plonk = RelaxedPlonkInstance::<C1>::new(0, 0, 0);
+
+        let bn_chip = BigUintMulModChip::<Base>::new(
+            config
+                .into_smaller_size::<{ big_uint_mul_mod_chip::MAIN_GATE_T }>()
+                .unwrap(),
+            LIMB_WIDTH,
+            LIMB_LIMIT,
+        );
+
+        for _round in 0..=NUM_OF_FOLD_ROUNDS {
+            let mut on_circuit_E_cell = None;
+            let cross_term_commits = random_curve_vec(&mut rnd);
+
+            // Run twice for setup & real run
```

Comment:
what is the difference when run twice here? Didn't get the meaning of "setup" part

Reply:
That's how halo2 works. In the first pass the advice (witness) is not filled, only fixed columns, and in the second pass the advice columns are filled. 

That's why closures are used as annotation type.

Reply:
I mean from this code `for _ in 0..=1`  is there any real difference that i=0 and i=1? It seems just run the same code twice because there is no index i but just "_".

Reply:
It runs down a completely identical code twice, but it's only on the second run that it actually fills witness. That's how halo 2 works.